pom.xml

```diff
@@ -129,16 +129,19 @@
         <dependency>
             <groupId>com.thoughtworks.xstream</groupId>
             <artifactId>xstream</artifactId>
+            <optional>true</optional>
```

Comment:
why should these be optional are they not used in the map code ?

Reply:
They are, but optional for consumers of the map rather than the map itself. They are still essentially `compile` ones for the map.

For me, this started when noticed maven libraries in dependencies. So I looked up where they are coming from and noticed that they are listed in an "optional" comment section here:
https://github.com/OpenHFT/Chronicle-Map/blob/develop/pom.xml#L127
```
        <!-- optional dependencies -->

        <dependency>
            <groupId>com.thoughtworks.xstream</groupId>
            <artifactId>xstream</artifactId>
        </dependency>

```

There are other pointers, like https://github.com/OpenHFT/Chronicle-Map/blob/develop/src/main/java/net/openhft/chronicle/map/JsonSerializer.java#L38
```
            "map.getAll(<file>) and map.putAll(<file>) methods require the JSON XStream serializer, " +
                    "we don't include these artifacts by default as some users don't require this functionality. " +
                    "Please add the following artifacts to your project\n" +
                    "<dependency>\n" +
                    " <groupId>xstream</groupId>\n" +
                    " <artifactId>xstream</artifactId>\n" +

```

This again tells me that these three are really supposed to be optional, but at the moment they are actually included by default, unlike what LOG says.

Reply:
My understanding is that it uses, some of these artifact which you have marked as option,  to serialise the maps header. So I’m reluctant to merge this pull request unless testing is done first.